src/ansible_navigator/steps.py

```diff
@@ -1,20 +1,26 @@
-"""steps here
+"""Step abstractions for actions.
 """
 from collections import deque
+from dataclasses import dataclass
+from dataclasses import field
 from typing import Any
 from typing import Callable
 from typing import Dict
+from typing import Generic
 from typing import List
 from typing import Optional
 from typing import Tuple
+from typing import TypeVar
 from typing import Union
 
 
 class Step:
-
     # pylint: disable=too-many-instance-attributes
     # pylint: disable=too-many-arguments
-    """One step in the flow of things"""
+    """One step in the flow of things.
+
+    This MappingStep below should be used for all new actions.
```

Comment:
Is `MappingStep` some sort of an identifier? Maybe highlight it, then?
```suggestion
    This ``MappingStep`` below should be used for all new actions.
```
OTOH, I don't see it existing in this module. Is it a typo?

Reply:
Good catch, it was intended to be TypedStep, the name of the new step type changed along the way.